src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+
+    if (arguments.length === 2) {
+      for (let i = 0; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
+    }
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+
+      for (let i = 1; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
+    }
```

Comment:
don't repeat yourself with loop. try to create a variable for starting index and change it altogether with your  accumulator if `arguments.length < 2`
```suggestion
    if (arguments.length  < 2) {
       accumulator = this[0];
       ...here you should also change your starting index
    }

    for (let i = your starting index variable; i < this.length; i++) {
      accumulator = callback(accumulator, this[i], i, this);
    }
```